optuna/pruners/percentile.py

```diff
@@ -93,12 +98,29 @@ def prune(self, study, trial):
         if step is None:
             return False
 
-        if step <= self.n_warmup_steps:
```

Comment:
Note that this assumed 1-based-indexing of reported steps. I think it's worth reconsidering this to be 0-based as it's Python and makes working with data/algorithms (e.g. the algorithm in this PR) more simple.

Reply:
This is probably better addressed in a separate issue. I aligned to the implementation to assume 1-based indexing.

Reply:
> I think it's worth reconsidering this to be 0-based as it's Python and makes working with data/algorithms (e.g. the algorithm in this PR) more simple.

note: Change to 0-based might require pruning callbacks in `optuna.integration` to be updated as different frameworks have different `base_index`.